internal/host/plugin/repository_host_catalog.go

```diff
@@ -454,6 +458,34 @@ func (r *Repository) UpdateCatalog(ctx context.Context, c *HostCatalog, version
 				recordUpdated = true
 			}
 
+			if needSetSync {
+				// We also need to mark all host sets in this catalog to be
+				// synced as well.
+				setsForCatalog, _, err := r.getSets(ctx, "", returnedCatalog.PublicId)
+				if err != nil {
+					return errors.Wrap(ctx, err, op, errors.WithMsg("unable to get sets for host catalog"))
+				}
+
+				for _, set := range setsForCatalog {
+					newSet := set.clone()
+					newSet.NeedSync = true
+					var msg oplog.Message
+					n, err := w.Update(ctx, newSet, []string{"NeedSync"}, []string{}, db.NewOplogMsg(&msg))
+					if err != nil {
+						return errors.Wrap(ctx, err, op, errors.WithMsg("unable to update host set"))
+					}
+
+					if n > 1 {
+						return errors.New(ctx, errors.MultipleRecords, op, fmt.Sprintf("expected no more than 1 host set to be updated while flagging host set id %q for synchronization, got %d", newSet.PublicId, n))
+					}
+
+					msgs = append(msgs, &msg)
+					if len(setsForCatalog) != 0 {
```

Comment:
Either remove the if statement (since inside this loop the `setsForCatalog` will always have a length > 0).  Or move it to after the for loop.  Maybe you meant to only set this if n ==  1 ?

Reply:
Fixed!